lib/i18n/backend/base.rb

```diff
@@ -163,7 +163,7 @@ def resolve(locale, object, subject, options = EMPTY_HASH)
         #   not standard with regards to the CLDR pluralization rules.
         # Other backends can implement more flexible or complex pluralization rules.
         def pluralize(locale, entry, count)
-          return entry unless entry.is_a?(Hash) && count
+          return entry unless entry.is_a?(Hash) && count && entry.values.select{|v| v.is_a?(Hash)}.size == 0
```

Comment:
Perhaps:

```
entry.values.none? { |v| v.is_a?(Hash) }
```

Reply:
Sure

Reply:
To help with archeology:

This implicitly changed the behaviour of pluralization contexts with nested hashes:

```yaml
en:
  stars:
    one: "%{count} star"
    other: "%{count} stars"
    special:
      one: "%{count} special star"
      other: "%{count} special stars"
```

Before:

```ruby
> I18n.t("stars", count: 1)
"1 star"
```

After:

```ruby
> I18n.t("stars", count: 1)
{:one=>"one", :other=>"other", :special=>{:one=>"one special star", :other=>"other special star"}}
```

Which was reported as https://github.com/ruby-i18n/i18n/issues/510, and was reverted in the case of `Backend::Pluralization` in https://github.com/ruby-i18n/i18n/commit/1b5e34553003ca3b42b842769e86c98d5e3b71d4, but not `Backend::Base`. The `Backend::Base` was reported as https://github.com/ruby-i18n/i18n/issues/514.